docs/src/snforge-advanced-features/conditional-compilation.md

```diff
@@ -0,0 +1,84 @@
+# Conditional Compilation
+
+It is possible to build some contracts solely for testing purposes.
+This can be achieved by leveraging [Scarb features](https://docs.swmansion.com/scarb/docs/reference/conditional-compilation.html#features).
+Configuration in `Scarb.toml` is done in the same manner as described in the Scarb documentation.
+Additionally, for utilizing features the `snforge test` command exposes the following flags, aligned with `scarb` flags:
+[--features](../appendix/snforge/test.md#-f---features-features),
+[--all-features](../appendix/snforge/test.md#--all-features) and [--no-default-features](../appendix/snforge/test.md#--no-default-features).
+
+## Contracts
+
+Firstly, define a contract in the `src` directory with a `#[cfg(feature: '<FEATURE_NAME>')]` attribute:
+
+```rust
+#[starknet::interface]
+trait IMockContract<TContractState> {
+    fn response(self: @TContractState) -> u32;
+}
+
+#[starknet::contract]
+#[cfg(feature: 'snforge_test_only')]
```

Comment:
A comment here explaining that the name can be `anything` could be useful. Or use a different feature name, that doesn't look like a build in thing in forge.

Reply:
Renamed to `enable_for_tests`